test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,207 @@
+		cmd := exec.Command("kubectl", "--kubeconfig", clusterProxy.GetKubeconfigPath(), "apply", "-k", kustomization) //nolint:gosec
+		output, err := cmd.CombinedOutput()
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("%s", output))
+
+		bmoDeployment := &v1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "baremetal-operator-controller-manager",
+				Namespace: "baremetal-operator-system",
+			},
+		}
+		// Wait for it to become available
+		framework.WaitForDeploymentsAvailable(ctx, framework.WaitForDeploymentsAvailableInput{
+			Getter:     clusterProxy.GetClient(),
+			Deployment: bmoDeployment,
+		}, "5m")
+		// Set up log watcher
+		framework.WatchDeploymentLogsByName(ctx, framework.WatchDeploymentLogsByNameInput{
+			GetLister:  clusterProxy.GetClient(),
+			Cache:      clusterProxy.GetCache(ctx),
+			ClientSet:  clusterProxy.GetClientSet(),
+			Deployment: bmoDeployment,
+			LogPath:    filepath.Join(artifactFolder, "logs", bmoDeployment.GetNamespace()),
+		})
+	}
+
+	if e2eConfig.GetVariable("DEPLOY_IRONIC") != "false" {
+		// Install Ironic
+		By("Installing Ironic")
+		kustomization := e2eConfig.GetVariable("IRONIC_KUSTOMIZATION")
+		cmd := exec.Command("kubectl", "--kubeconfig", clusterProxy.GetKubeconfigPath(), "apply", "-k", kustomization) //nolint:gosec
+		output, err := cmd.CombinedOutput()
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("%s", output))
+
+		ironicDeployment := &v1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "baremetal-operator-ironic",
+				Namespace: "baremetal-operator-system",
+			},
+		}
+		// Wait for it to become available
+		framework.WaitForDeploymentsAvailable(ctx, framework.WaitForDeploymentsAvailableInput{
+			Getter:     clusterProxy.GetClient(),
+			Deployment: ironicDeployment,
+		}, "10m")
+		// Set up log watcher
+		framework.WatchDeploymentLogsByName(ctx, framework.WatchDeploymentLogsByNameInput{
+			GetLister:  clusterProxy.GetClient(),
+			Cache:      clusterProxy.GetCache(ctx),
+			ClientSet:  clusterProxy.GetClientSet(),
+			Deployment: ironicDeployment,
+			LogPath:    filepath.Join(artifactFolder, "logs", ironicDeployment.GetNamespace()),
+		})
+	}
+
+	return []byte(strings.Join([]string{clusterProxy.GetKubeconfigPath()}, ","))
+}, func(data []byte) {
+	// Before each parallel node
+	parts := strings.Split(string(data), ",")
+	Expect(parts).To(HaveLen(1))
+
+	kubeconfigPath := parts[0]
+	scheme := runtime.NewScheme()
+	framework.TryAddDefaultSchemes(scheme)
+	bmov1.AddToScheme(scheme)
+
+	e2eConfig = LoadE2EConfig(configPath)
+	clusterProxy = framework.NewClusterProxy("bmo-e2e", kubeconfigPath, scheme)
+})
+
+// Using a SynchronizedAfterSuite for controlling how to delete resources shared across ParallelNodes (~ginkgo threads).
+// The kubernetes cluster is shared across all the tests, so it should be deleted only after all ParallelNodes completes.
+// The local clusterctl repository is preserved like everything else created into the artifact folder.
```

Comment:
would we have clusterctl repo here?  possibly not

Reply:
Good point! Copied a bit too much from CAPI.